evm/src/main/java/org/hyperledger/besu/evm/internal/FlexStack.java

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright contributors to Hyperledger Besu
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.evm.internal;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import java.lang.reflect.Array;
+import java.util.Arrays;
+import java.util.Objects;
+
+/**
+ * An operand stack for the Ethereum Virtual machine (EVM). The stack grows 32 entries at a time if
+ * it expands past the top of the allocated stack, up to maxSize.
+ *
+ * <p>The operand stack is responsible for storing the current operands that the EVM can execute. It
+ * is assumed to have a fixed size.
+ *
+ * @param <T> the type parameter
+ */
+public class FlexStack<T> {
+
+  private static final int INCREMENT = 32;
+
+  private T[] entries;
+
+  private final int maxSize;
+  private int currentCapacity;
+
+  private int top;
+
+  /**
+   * Instantiates a new Fixed stack.
+   *
+   * @param maxSize the max size
+   * @param klass the klass
+   */
+  @SuppressWarnings("unchecked")
+  public FlexStack(final int maxSize, final Class<T> klass) {
+    checkArgument(maxSize >= 0, "max size must be non-negative");
```

Comment:
does it make sense to allow zero?

Reply:
Thats been in the codebase since before I started working in it.  I can adjust it.